rasa/shared/nlu/training_data/training_data.py

```diff
@@ -679,38 +678,54 @@ def can_train_nlu_model(self) -> bool:
         ]
         return not any([len(lst) > 0 for lst in lists_to_check])
 
-    def divide_into_chunks(self, max_size: int = 2) -> List["TrainingDataChunk"]:
+    def divide_into_chunks(self, num_chunks: int) -> List["TrainingDataChunk"]:
         """Divides the training data into smaller chunks.
 
         Each chunk should be a good representation of the complete dataset. E.g. it
         should not just contain examples of one intent, but instead match the
         distribution of the complete dataset.
 
         Args:
-            max_size: The maximum size (in MB) of one chunk.
+            num_chunks: The total number of chunks into which the training data should be broken.
 
         Returns:
             A list of all training data chunks.
         """
-        pass
+        all_chunks = []
+        data_to_chunk = copy.deepcopy(self)
 
+        for chunk_index in range(num_chunks):
 
-class TrainingDataChunk:
-    """Holds a portion of the complete TrainingData."""
+            chunk_size_fraction = 1 / (num_chunks - chunk_index)
+            current_chunk, leftover_examples = data_to_chunk.split_nlu_examples(
+                1 - chunk_size_fraction
+            )
 
-    def __init__(
-        self,
-        training_examples: List[Message],
-        responses: Optional[Dict[Text, List[Dict[Text, Any]]]] = None,
-    ) -> None:
-        """Initialize a training data chunk.
+            if chunk_index == num_chunks - 1:
+                # There can be a small number of leftover examples
+                # during the last bit of chunking because of rounding errors.
+                current_chunk.extend(leftover_examples)
+
+            # update the data to chunk in next iteration
+            data_to_chunk = TrainingData(
+                leftover_examples,
+                responses=data_to_chunk._needed_responses_for_examples(
+                    leftover_examples
+                ),
+            )
+            all_chunks.append(
+                TrainingDataChunk(
+                    current_chunk,
+                    responses=data_to_chunk._needed_responses_for_examples(
+                        current_chunk
+                    ),
+                )
+            )
+        return all_chunks
 
-        Args:
-            training_examples: List of messages representing the training data.
-            responses: List of responses.
-        """
-        self.training_examples = training_examples
-        self.responses = responses
+
+class TrainingDataChunk(TrainingData):
```

Comment:
I am super out of context for this unfortunately, but I'm interested into why a `TrainingDataChunk` is a subclass of `TrainingData` rather than `TrainingData` having a reference to a list of `TrainingDataChunk`.

Reply:
`TrainingDataChunk` will hold a subset of `training_examples` and `responses` on which a lot of helper methods that `TraningData` class has can be used. For e.g., lazy properties.